unit_tests/src/test_system_error.cpp

```diff
@@ -0,0 +1,78 @@
+#include <catch2/catch.hpp>
+
+#include "irods/rodsErrorTable.h"
+#include "irods/system_error.hpp"
+
+#include <fmt/format.h>
+
+#include <limits>
+#include <string>
+#include <string_view>
+
+TEST_CASE("irods_category")
+{
+    using namespace std::string_view_literals;
+    constexpr auto ec = USER_FILE_DOES_NOT_EXIST;
+    const auto errc = irods::experimental::make_error_code(ec);
+    CHECK(errc.value() == ec);
+    CHECK(irods::experimental::get_irods_error_code(errc) == USER_FILE_DOES_NOT_EXIST);
+    CHECK(irods::experimental::get_errno(errc) == 0);
+    CHECK(errc.message() == "USER_FILE_DOES_NOT_EXIST");
+    CHECK(errc.default_error_condition().category().name() == "iRODS"sv);
+}
+
+TEST_CASE("irods_category reports embedded errno value")
+{
+    using namespace std::string_view_literals;
+    constexpr auto ec = USER_FILE_DOES_NOT_EXIST - 120; // 120 = EINTR
+    const auto errc = irods::experimental::make_error_code(ec);
+    CHECK(errc.value() == ec);
+    CHECK(irods::experimental::get_irods_error_code(errc) == USER_FILE_DOES_NOT_EXIST);
+    CHECK(irods::experimental::get_errno(errc) == 120);
+    CHECK(errc.message() == "USER_FILE_DOES_NOT_EXIST [errno=120]");
+    CHECK(errc.default_error_condition().category().name() == "iRODS"sv);
+}
+
+TEST_CASE("irods_category with positive value")
+{
+    using namespace std::string_view_literals;
+    constexpr auto ec = -USER_FILE_DOES_NOT_EXIST;
+    const auto errc = irods::experimental::make_error_code(ec);
+    CHECK(errc.value() == ec);
+    CHECK(irods::experimental::get_irods_error_code(errc) == -USER_FILE_DOES_NOT_EXIST);
+    CHECK(irods::experimental::get_errno(errc) == 0);
+    CHECK(errc.message() == fmt::format("Unknown error {}", ec));
+    CHECK(errc.default_error_condition().category().name() == "iRODS"sv);
+}
+
+TEST_CASE("irods_category with positive value with embedded errno value")
+{
+    using namespace std::string_view_literals;
+    constexpr auto ec = -USER_FILE_DOES_NOT_EXIST + 120; // 120 = EINTR
+    const auto errc = irods::experimental::make_error_code(ec);
+    CHECK(errc.value() == ec);
+    CHECK(irods::experimental::get_irods_error_code(errc) == -USER_FILE_DOES_NOT_EXIST);
+    CHECK(irods::experimental::get_errno(errc) == 120);
```

Comment:
How do we feel about the `get_` prefix?

Does `extract_` seem better? I feel it expresses intent better at the cost of more characters.

I originally planned to use `strip_`, but that inverts the meaning and takes more effort to understand. For example, `strip_errno(errc)` means _give me the iRODS error code without the errno_.

Reply:
`extract_` seems the least confusing and the most specific.

Reply:
`extract_` it is.

Reply:
I personally like `get_` more because `extract_` implies a modification to the object (by definition, "extract" means "to remove or take out"). What intent does `extract_` express that `get_` is not expressing? I'm not strongly opinionated on this, just want to understand the appeal :)

Reply:
'implies a modification'... oh, does it?   if so, then `get_` is more correct / fine with me.

Reply:
Well, it implies a modification to me. There may be some standard usage of "extract" of which I am currently unaware

Reply:
I didn't know that was the meaning of extract. I'll definitely remember that for next time.

`get` can mean anything, but does work. I was trying to express that the function will return an embedded value without modification.

If there is a word that can convey that, that would be nice.

Reply:
`get` seems great.

Reply:
Moving forward with `get`.

Reply:
Done.